### DISCUSSION TITLE
What's the plan re: Move admin to more recent FontAwesome i.e. 6.x?

### DISCUSSION THREAD

**Discussion by neekfenwick** [zencart/zencart#5763] [Ideas]
I see ZC 158 still defaults to using FontAwesome 4.7.0, and the latest version at time of writing is 6.4.0.  What are people's general opinions about moving to a more recent version?

Is this situation a case of "if you want it, make a PR" or is there some stronger push back such as "too much trouble, don't bother"?

Since version 4 they've changed to using a different namespace class for their icons and renamed a few groups of icons, for example the class "fa" is now one of "far" (regular) "fas" (solid) etc., and some icon names with shapes in them such as "fa-info-circle" are now "fa-circle-info" so the first part "fa-circle-" is common across all circular icons.

There is back-compat in the 6.x release, so using "fa fa-info-circle" does still work, but the preferred new syntax would be "fas fa-circle-info".  There is also a shims CSS that fixes obsolete names but it would be preferable not to have to use it.

As I see it, simple pros and cons are:

Pros:
- ZC stays up to date with this third party library.
- Add-on devs using FontAwesome can use the newer naming scheme.

Cons:
- Breakage with any add-ons that currently use fontawesome.
- FontAwesome 7 will be along at some point and make 6 obsolete :P

I'm biased because our site has switched to using a licenced Kit build of FontAwesome 6 to minimise load time and we've converted all our catalog files to use the new release, but the admin side still loads the older 4.x release and switching between them can be a pain.

I tried dropping v6 in with this change in admin_html_head.php and renaming the bundled font-awesome.min.css so it's not found:

```
<?php if (file_exists($value = 'includes/css/font-awesome.min.css')) { ?>
    <link rel="stylesheet" href="<?php echo $value; ?>">
<?php } else { ?>
    <!-- REMOVE OLD 4.x CDN LINK link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous" -->
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/v4-shims.css" crossorigin="anonymous">
<?php } ?>
```

This works fine but it would be nice not to have to use the v4-shims.css.  It's there because of completely obsolete icons e.g. category_product_listing uses fa-trash-o which now is just fa-trash.  For reference, https://fontawesome.com/docs/web/setup/upgrade/upgrade-from-v4 lists obsolete icon names.

**Comment by scottcwilson** [comment]
My vote would be to upgrade, add the shim, and add a mechanism to suppress loading the shim (e.g. through a site_specific file of some sort).

**Reply by neekfenwick** [reply]
Would it make more sense to not load the shim by default, and add a mechanism to load it? Core ZC would not need the shim because we'd upgrade all the icon usage, it's only potential use of add-ons that would cause a site to need the shim.
Edit: Since I'm a bit new to ZC modifications, what mechanism do you suggest? A new `configuration` entry with a true|false value?  I don't see a `configuration_group` that's really suitable for it to live in, maybe just `configuration_group_id` 1 'My Store'.

**Comment by neekfenwick** [comment]
Question about GitHub practises: If I was to prepare a PR for this Discussion, should I make an Issue for the work, or just create a branch named after this Discussion and PR based on that?

I could also push what I've done for the email templating work but wasn't sure if I should create an Issue first or not.

**Comment by scottcwilson** [comment]
For the FA change - just create a branch.  Same for email, but because email is so bigger please update the discussion to say where you are.

**Reply by neekfenwick** [reply]
Thanks.  See https://github.com/zencart/zencart/pull/5845

**Comment by lat9** [comment]
Closing this discussion, since #5845 has been merged.
